src/Network/JSONApi/Resource.hs

```diff
@@ -77,7 +77,7 @@ instance (NFData a) => NFData (Resource a)
 {- |
 A typeclass for decorating an entity with JSON API properties
 -}
-class (ToJSON a, FromJSON a) => ResourcefulEntity a where
+class ResourcefulEntity a where
```

Comment:
@gwils can you explain why making this type class more general is a good idea? Can we not just put `ToJSON` to the list?

Reply:
I think it's useful to not require constraints that we're not using. We can move these constraints closer to the use site.

In the same way that we're using only `ToJSON, ResourcefulEntity` in our project, I can imagine another project where someone is only using `FromJSON`, so I'd rather assume neither.